code/code/spec/spec_objs_weapons.cc

```diff
@@ -2748,3 +2748,197 @@ int glacialWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
   return FALSE;
 }
 
+// Summary of this weapon proc
+// - lightning damage proc
+// - 1.75% chance of wither limb, with a white lightning theme (withers the limb due to electric damage)
+// - Flavor text proc of arching blue electricity from the weapon
+
+int lightningboltWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
+{
+char limb[80];
+
+int rc, dam = 0, chance;
+
+TBeing *ch;
+
+sstring buf;
+
+wearSlotT slot;
+
+if (!(ch = dynamic_cast<TBeing *>(o->equippedBy)))
+return FALSE; // weapon not equipped (carried or on ground)
+// Flavor text pulse
+if (cmd == CMD_GENERIC_PULSE)
+{ 
+if (!::number(0,150)){
+act("<w>Crackles of <b>blue electricity<z> from $n's $o arch into the $g.<1>",
+0, ch, o, 0, TO_ROOM);
+
+act("<w>Your $o crackles with <b>blue electricity<z> arching into the $g.<1>",
+0, ch, o, 0, TO_CHAR);
+
+}
+}
+
+// Combat procs
+if(cmd == CMD_OBJ_HIT)
+{
+// Slightly lowering chance to proc to balance with bone break/unmaker
+// Since this has extra damage components
+chance = ::number(1,57);
+
+if (!o || !vict)
+return FALSE;
+
+// Witherlimb proc + 10-20 damage if successful (max once/fight)
+
+if (chance == 1) {
+if(!ch->canWither(vict, SILENT_YES))
+return false;
+
+bool ok = false;
+	
+bool found = false;
+
+// Initial checks to prevent issues and for balancee reasons
+for (slot = MIN_WEAR; slot < MAX_WEAR; slot++) {
+if (notBreakSlot(slot, false)) // same ones, right?
+continue;
+
+if (!vict->slotChance(slot))
+continue;
+
+found |= (vict->isLimbFlags(slot, PART_MISSING));
+
+ok = true;
+}
+
+if (found || !ok) 
+return false;
+
+// Accounting for electricity immunity since this is electricity-based wither limb
+if (vict->isImmune(IMMUNE_ELECTRIC, slot))
+return false;
+
+// Get a random limb slot and make sure it's valid
+for (slot = pickRandomLimb();; slot = pickRandomLimb()) {
+if (notBreakSlot(slot, true))
+continue;
+
+if (!vict->slotChance(slot))
+continue;
+
+break;
+
+}
+
+if (!vict->hasPart(slot)) {
+vlogf(LOG_COMBAT, format("BOGUS SLOT trying to be made PART_USELESS: %d on %s") % 
+slot % vict->getName());
+
+return FALSE;
+
+}
+
+if (!vict->roomp) {
+vlogf(LOG_COMBAT, format("!roomp for target (%s) when trying to make PART_USELESS.") % vict->getName());
+return FALSE;
+
+}
+
+vict->setLimbFlags(slot, PART_MISSING);
+
+TThing *t;
+
+if ((t = vict->unequip(slot)))
+*(vict->roomp) += *t;
+
+for (wearSlotT j=MIN_WEAR; j < MAX_WEAR; j++) {
+if (!vict->hasPart(j))
+continue;
+
+if (!vict->limbConnections(j)) {
```

Comment:
Is this file mostly copy-paste? Maybe we could consolidate the code with bonebreak or lightsaber or such, passing in the specific flavor strings and chances and whatnot?